rkt/fetch.go

```diff
@@ -47,9 +58,111 @@ func init() {
 	cmdFetch.Flags().Var((*appAsc)(&rktApps), "signature", "local signature file to use in validating the preceding image")
 	cmdFetch.Flags().BoolVar(&flagStoreOnly, "store-only", false, "use only available images in the store (do not discover or download from remote URLs)")
 	cmdFetch.Flags().BoolVar(&flagNoStore, "no-store", false, "fetch images ignoring the local store")
+	cmdFetch.Flags().IntVar(&flagP2pDuration, "p2p-duration", 10, "p2p continue service duration (minutes) after download finished")
+}
+
+func p2pFetch(args []string) int {
+	if len(args) < 1 {
+		stderr("fetch: must provide tottent file.")
+		return 1
+	}
+	
+	d:=strconv.Itoa(flagP2pDuration)
+	//start p2p client		
+	var cmd *exec.Cmd
+	cmd = exec.Command("./torrent", args[0],d)
+	err := cmd.Start()
+	if err != nil {
+		fmt.Printf("start p2p process err,%s\n",err)
+		return 1
+	}
+	
+	pid:=cmd.Process.Pid
+
+	//handle Ctrl+C signal
+	go func(){	
+		c := make(chan os.Signal, 1)
+	    signal.Notify(c, os.Interrupt, os.Kill)
+	
+	    <-c
+		if err = syscall.Kill(pid, 9); err != nil {
+			fmt.Printf("kill p2p client process err,%s\n",err)
+		}
+   }()
+
+	//wait for p2p client start
+	time.Sleep(time.Second * time.Duration(7))
+	
+	//connet to p2p client process and get download rate
+	client, err := rpc.DialHTTP("tcp", "127.0.0.1:1234")
+	if err != nil {
+		fmt.Printf("connet to p2p client err,%s\n", err)
+		return 1
+	}
+	reply := make([]string, 1)
+	err = client.Call("Download.GetDownloadTotalSize", struct{}{}, &reply)
+	if err != nil {
+		fmt.Printf("get download total size err,%s\n", err)
+		return 1
+	}
+	totalSize:=reply[0]	
+	
+	err = client.Call("Download.GetDownloadFile", struct{}{}, &reply)
+	if err != nil {
+		fmt.Printf("get download file err,%s\n", err)
+		return 1
+	}
+	aciImage := reply[0]	
+
+	//get rate for download
+	for {
+		err = client.Call("Download.GetDownloadRate", struct{}{}, &reply)
+		if err != nil {
+			fmt.Printf("get download rate err,%s\n", err)
+			return 1
+		}
+		fmt.Printf("\rtotal Size:%sKB, downloaded:%s",totalSize,reply) 
+		if reply[0]=="100.00%"{
+			break
+		}
+		time.Sleep(time.Second * time.Duration(5))
+	}
+	
+	//save aci to rkt store
+	s, err := store.NewStore(globalFlags.Dir)
+	if err != nil {
+		fmt.Printf("cannot open store: %v", err)
+		return 1
+	}
+	aciFile, err := os.Open(aciImage)
+	if err != nil {
+		fmt.Printf("opening ACI file %s failed: %v", aciImage, err)
+		return 1
+	}
+	key, err := s.WriteACI(aciFile, true)
+	if err != nil {
+		fmt.Printf("write ACI file failed: %v", err)
+		return 1
+	}
+	fmt.Println(key)
+		
+	return 0
 }
 
 func runFetch(cmd *cobra.Command, args []string) (exit int) {
+	
+	//check if use p2p download
+	b:=func(fileName string) bool{
+    	_, err := os.Stat(fileName)
+    	return err == nil || os.IsExist(err)
+	}(args[0])
```

Comment:
Instead commenting unnamed func - please extract it and properly name it. Also please try to follow normal convention of white space using.
Please also change name or variable accepting return value into something more descriptive than `b`.